android/filament-android/src/main/java/com/google/android/filament/android/UiHelper.java

```diff
@@ -181,39 +182,71 @@ public int getDesiredHeight() {
         return mDesiredHeight;
     }
 
+    /**
+     * Returns true if the render target is opaque.
+     */
+    public boolean isOpaque() {
+        return mOpaque;
+    }
+
+    /**
+     * Controls whether the render target (SurfaceView or TextureView) is opaque or not.
+     * The render target is considered opaque by default.
+     *
+     * Must be called before calling {@link #attachTo(SurfaceView)}
+     * or {@link #attachTo(TextureView)}.
+     *
+     * @param opaque Indicates whether the render target should be opaque. True by default.
+     */
+    public void setOpaque(boolean opaque) {
+        mOpaque = opaque;
+    }
+
+    public long getSwapChainFlags() {
+        return isOpaque() ? SwapChain.CONFIG_DEFAULT : SwapChain.CONFIG_TRANSPARENT;
+    }
+
     /**
      * Associate UiHelper with a SurfaceView.
      *
      * As soon as SurfaceView is ready (i.e. has a Surface), we'll create the
      * EGL resources needed, and call user callbacks if needed.
      */
     public void attachTo(SurfaceView view) {
-        final SurfaceHolder.Callback callback = new SurfaceHolder.Callback() {
-            @Override
-            public void surfaceCreated(SurfaceHolder holder) {
-                if (LOGGING) Log.d(LOG_TAG, "surfaceCreated()");
-                createSwapChain(holder.getSurface());
+        if (attach(view)) {
+            if (!isOpaque()) {
+                view.setZOrderOnTop(true);
```

Comment:
This is a just a helper and not required by the engine at all. I'm also not sure what would be the point of creating a transparent surface view that doesn't sit on top of the activity. This can be revisited as needed later.

Reply:
Yes, you are right. I think it is ok for now.